### DISCUSSION TITLE
What do the Magpie Directories do?

### DISCUSSION THREAD

**Discussion by lordjoe** [llnl/magpie#331] [General]

**Comment by lordjoe** [comment]
I presume examples and doc are not needed in a working implementation. Is patches needed. What about testsuite? I hit test-generate.sh and an hour later it is still running. What did I just do? Are tests useful for ordinary users? if so what do they test and how should they be run?

**Comment by chu11** [comment]
> Is patches needed.

After you apply the patches to projects, no.  See `README.hadoop`, `README.spark`, etc. to see if patches are required for your project/environment.

> What about testsuite?

No

> I hit test-generate.sh and an hour later it is still running. What did I just do?

You generated test batch files that can be submitted to the scheduler/resource manager for testing.

> Are tests useful for ordinary users?

They could provide examples of how to configure Magpie in alternate ways, but I doubt its useful for users.

> if so what do they test

They test a wide variety of configurations.  Different projects versions, filesystems, schedulers, etc.

>  and how should they be run?

See testsuite documentation for more info: https://github.com/LLNL/magpie/blob/master/doc/README.testsuite
